core/new-gui/src/app/common/type/workflow.ts

```diff
@@ -1,12 +1,38 @@
-export interface Workflow extends Readonly<{
+import { Breakpoint, OperatorLink, OperatorPredicate, Point } from '../../workspace/types/workflow-common.interface';
+
+/**
+ * CachedWorkflow is used to store the information of the workflow
+ *  1. all existing operators and their properties
+ *  2. operator's position on the JointJS paper
+ *  3. operator link predicates
+ *
+ * When the user refreshes the browser, the CachedWorkflow interface will be
+ *  automatically cached and loaded once the refresh completes. This information
+ *  will then be used to reload the entire workflow.
+ *
+ */
+export interface WorkflowInfo {
+  operators: OperatorPredicate[];
+  operatorPositions: { [key: string]: Point };
+  links: OperatorLink[];
+  breakpoints: Record<string, Breakpoint>;
+}
+
+export interface Workflow {
   name: string;
-  wid: number;
-  content: string;
-  creationTime: string;
-  lastModifiedTime: string;
-}> {
+  wid: number | null;
+  content: WorkflowInfo;
+  creationTime: number;
+  lastModifiedTime: number;
 }
 
+export function parseWorkflowInfo(workflow: Workflow) {
```

Comment:
move this to somewhere else, make it a public static function of a service, this file should only contain interface declarations

Reply:
can it be under the interface declaration?